plasma_framework/contracts/src/exits/payment/controllers/PaymentChallengeIFENotCanonical.sol

```diff
@@ -154,4 +171,53 @@ library PaymentChallengeIFENotCanonical {
 
         return utxoPos.value;
     }
+
+    function verifyCompetingTxFinalized(
+        Controller memory self,
+        PaymentInFlightExitRouterArgs.ChallengeCanonicityArgs memory args
+    )
+        private
+        view
+        returns (uint256)
+    {
+        // default to infinite low priority position
+        uint256 competitorPosition = ~uint256(0);
+
+        UtxoPosLib.UtxoPos memory competingTxUtxoPos = UtxoPosLib.UtxoPos(args.competingTxPos);
+        uint256 competingTxType = WireTransaction.getTransactionType(args.competingTx);
+        uint8 protocol = self.framework.protocols(competingTxType);
+
+        if (args.competingTxPos == 0) {
+            require(protocol == Protocol.MORE_VP(), "Competing tx without position must be a more vp tx");
+        } else {
+            IOutputGuardHandler outputGuardHandler = self
+                                                    .outputGuardHandlerRegistry
+                                                    .outputGuardHandlers(args.outputType);
+
+            require(address(outputGuardHandler) != address(0), "Failed to get the outputGuardHandler of the output type");
+
+            WireTransaction.Output memory output = WireTransaction.getOutput(args.inputTx, args.inFlightTxInputIndex);
+            OutputGuardModel.Data memory outputGuardData = OutputGuardModel.Data({
```

Comment:
okay, I split the logic to `verifyOutputTypeAndPreimage` and also I makes sure it would be run for all cases. (Currently this check is only run when there is position given)